### PR TITLE
Make File/Path calls more robust

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

**Comment by CookiePLMonster** [comment]
There are a few more instances in the code where throwing `fs` methods are used, for example `fs::absolute` in `ResolveFilenameForCapture` and `fs::recursive_directory_iterator` in `FileWatcher::FileWatcher`.

Looks like Ctrl+F'ing for `fs::` in the codebase can find them all, as `using namespace fs;` is never used.

**Comment by Gymnasiast** [comment]
The constructor for FileWatcher can throw exceptions anyway, and the code calling the constructor is wrapped in a try-catch.

I skipped fs::absolute as I want to move that to Path::GetAbsolute once that is cleaned up, along with platform_get_absolute_path().

File: src/openrct2/core/File.cpp [hunk 1]
```diff
@@ -27,7 +27,9 @@ namespace File
     {
         fs::path file = u8path(path);
         log_verbose("Checking if file exists: %s", std::string(path).c_str());
-        return fs::exists(file);
+        std::error_code ec;
+        const auto result = fs::exists(file, ec);
+        return result && ec.value() == 0;
```
Comment (tupaschoal) [inline src/openrct2/core/File.cpp:32]: `std::error_code` does have an `operator bool()` implementation, so technically you could do:
```suggestion
        return result && !ec;
```

I'll leave it up to you to decide. If you follow my suggestion, make sure you update on the rest of the diff.

**Review by tupaschoal — APPROVED** [review]

File: src/openrct2/core/File.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/core/File.cpp:32]: You know me, I like being more explicit ;-)

Comment (tupaschoal) [reply src/openrct2/core/File.cpp:32]: Yeah, I thought so :PP

**Pull request merged**